server/mscalendar/availability_test.go

```diff
@@ -430,6 +430,105 @@ func TestRetrieveUsersToSyncIndividually(t *testing.T) {
+		s.EXPECT().LoadUser(testUser.MattermostUserID).Return(testUser, nil).Times(2)
+		s.EXPECT().LoadUser(testUser2.MattermostUserID).Return(testUser2, nil)
+
+		events := []*remote.Event{newTestEvent("1", "", "test")}
+		papi.EXPECT().GetMattermostUser(testUser.MattermostUserID)
+		r.EXPECT().MakeClient(gomock.Any(), testUser.OAuth2Token).Return(client)
+		c.EXPECT().GetEventsBetweenDates(testUser.Remote.ID, gomock.Any(), gomock.Any()).Return(events, nil)
+
+		m := New(e, "").(*mscalendar)
+		jobSummary := &StatusSyncJobSummary{}
+
+		users, responses, err := m.retrieveUsersToSync(userIndex, jobSummary, true)
+		require.NoError(t, err)
+		require.Equal(t, users, []*store.User{testUser})
+		require.Equal(t, responses, []*remote.ViewCalendarResponse{{
+			RemoteUserID: testUser.Remote.ID,
+			Events:       events,
+		}})
+	})
+
+	t.Run("two users should be synced", func(t *testing.T) {
+		testUser := newTestUserNumbered(1)
+		testUser.Settings.UpdateStatus = true
+		testUser.Settings.ReceiveReminders = true
+
+		testUser2 := newTestUserNumbered(2)
+		testUser2.Settings.UpdateStatus = true
+		testUser2.Settings.ReceiveReminders = true
+
+		userIndex := []*store.UserShort{
+			{
+				MattermostUserID: testUser.MattermostUserID,
+				RemoteID:         testUser.Remote.ID,
+				Email:            testUser.Remote.Mail,
+			},
+			{
+				MattermostUserID: testUser2.MattermostUserID,
+				RemoteID:         testUser2.Remote.ID,
+				Email:            testUser2.Remote.Mail,
+			},
+		}
+		ctrl := gomock.NewController(t)
+		defer ctrl.Finish()
+
+		e, client := makeStatusSyncTestEnv(ctrl)
+
+		c, r, _, s, papi := client.(*mock_remote.MockClient), e.Remote.(*mock_remote.MockRemote), e.Poster.(*mock_bot.MockPoster), e.Store.(*mock_store.MockStore), e.PluginAPI.(*mock_plugin_api.MockPluginAPI)
+		s.EXPECT().LoadUser(testUser.MattermostUserID).Return(testUser, nil).Times(2)
+		s.EXPECT().LoadUser(testUser2.MattermostUserID).Return(testUser2, nil).Times(2)
+
+		eventsUser1 := []*remote.Event{newTestEvent("1", "", "test")}
+		eventsUser2 := []*remote.Event{newTestEvent("2", "", "test2")}
+		papi.EXPECT().GetMattermostUser(testUser.MattermostUserID)
+		papi.EXPECT().GetMattermostUser(testUser2.MattermostUserID)
+		r.EXPECT().MakeClient(gomock.Any(), testUser.OAuth2Token).Return(client)
+		r.EXPECT().MakeClient(gomock.Any(), testUser2.OAuth2Token).Return(client)
+		c.EXPECT().GetEventsBetweenDates(testUser.Remote.ID, gomock.Any(), gomock.Any()).Return(eventsUser1, nil)
+		c.EXPECT().GetEventsBetweenDates(testUser2.Remote.ID, gomock.Any(), gomock.Any()).Return(eventsUser2, nil)
+
+		m := New(e, "").(*mscalendar)
+		jobSummary := &StatusSyncJobSummary{}
+
+		users, responses, err := m.retrieveUsersToSync(userIndex, jobSummary, true)
+		require.NoError(t, err)
+		require.Equal(t, users, []*store.User{testUser, testUser2})
+		require.Equal(t, responses, []*remote.ViewCalendarResponse{{
```

Comment:
Nit: The order of args should go `t, expected, actual` rather than `t, actual, expected`. This makes it so a failing test output will display the correct error message

Reply:
I always put the order wrong. 🤦